Lacework/lacework_module/lacework_connector.py

```diff
@@ -139,57 +139,55 @@ def get_response_by_timestamp(self) -> Any | None:
             )
 
             return None
+
         try:
-            return response.json()
+            result: dict[str, Any] = response.json()
+
+            return result
         except ValueError:
-            self.log(
-                message=("No more messages to forward"),
-                level="info",
-            )
+            self.log(message="No more messages to forward", level="info")
+
             return None
 
-    def _get_most_recent_timestamp_from_items(self, items: list[dict[Any, Any]]) -> float:
-        def _extract_timestamp(item: dict[Any, Any]) -> float:
-            RFC3339_STRICT_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
-            return datetime.strptime(item["endTime"], RFC3339_STRICT_FORMAT).replace(tzinfo=timezone.utc).timestamp()
+    @staticmethod
+    def parse_event_time(value: str) -> datetime:
+        RFC3339_STRICT_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+        return datetime.strptime(value, RFC3339_STRICT_FORMAT).replace(tzinfo=timezone.utc)
+
+    @staticmethod
+    def get_next_page_url(response: dict[str, Any]) -> str | None:
+        result: str | None = response.get("content", {}).get("paging", {}).get("urls", {}).get("nextPage")
 
-        later = 0.0
-        for item in items:
-            if _extract_timestamp(item) > later:
-                later = _extract_timestamp(item)
-        return float(later)
+        return result
 
     def forward_next_batches(self) -> None:
         """
         Successively queries the Lacework Central API while more are available
         and the current batch is not too big.
         """
         first_batch = self.get_response_by_timestamp()
-        if first_batch and first_batch != "":
-            next_page_url = first_batch.get("content", {}).get("paging", {}).get("urls", {}).get("nextPage")
-            items = first_batch.get("data", [])
-            last_date = self._get_most_recent_timestamp_from_items(items)
-            events_lag = int(time.time() - last_date)
-            EVENTS_LAG.labels(intake_key=self.configuration.intake_key).set(events_lag)
-            self.most_recent_date_seen = datetime.utcfromtimestamp(last_date)
-            messages = [orjson.dumps(message).decode("utf-8") for message in items]
-            self.log(message=f"Sending the first batch of {len(messages)} messages", level="info")
-            OUTCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc(len(messages))
-            self.push_events_to_intakes(events=messages)
+        if first_batch is not None:
+            next_page_url = self.get_next_page_url(first_batch)
+            first_batch_events = first_batch.get("data", [])
+
+            last_date = max([self.parse_event_time(item["endTime"]) for item in first_batch_events])
```

Comment:
We still uses `endTime` to save the last date, but we [uses `startTime` when requesting the alerts](https://github.com/SEKOIA-IO/automation-library/blob/main/Lacework/lacework_module/client/__init__.py#L39).
So, I think that the PR will not fix the issue (see Hypothesis in the ticket).